### PR TITLE
fix: headless job tracking in printToPDF

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Fixes headless job tracking for `contents.printToPDF()`, which was incorrectly/unintentionally removed in [a recent Chromium roll](https://github.com/electron/electron/pull/34993/files#diff-2b43b0cabaeb72a224a224d5e7611592bf1c5e02d890c7fe8d132e41bd39d6e5L132-L134).

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).

#### Release Notes

Notes: Fixed a potential issue when calling `webContents.printToPDF()`.

### CODE REVIEW

File: shell/browser/printing/print_view_manager_electron.cc [hunk 1]
```diff
@@ -169,6 +172,10 @@ void PrintViewManagerElectron::OnDidPrintWithParams(
   std::move(callback_).Run(kPrintSuccess,
                            base::RefCountedString::TakeString(&data));
 
+  headless_jobs_.erase(std::remove(headless_jobs_.begin(), headless_jobs_.end(),
```
Comment (zcbenz) [inline shell/browser/printing/print_view_manager_electron.cc:175]: There is a `base::Erase` helper in `cxx20_erase_vector.h` to make code more readable.

**Review by zcbenz — APPROVED** [review]

**Comment by deepak1556** [comment]
Does https://github.com/electron/electron/pull/36051 supersede this PR ?

**Comment by codebytere** [comment]
@deepak1556 similar but i'd prefer to merge this separately since they're addressing separate functional areas i'd say!

**Review by deepak1556 — APPROVED** [review]
+1 on `base::Erase`

**Pull request merged**